data_steward/validation/participants/identity_match.py

```diff
@@ -485,82 +572,333 @@ def _compare_birth_dates(person_id_set, rdr_birthdates, ehr_birthdates, match_va
+        )
+        # write email matches for hpo to table
+        writers.append_to_result_table(
+            site,
+            match_values,
+            project,
+            validation_dataset,
+            consts.EMAIL_FIELD
+        )
+
+    # validate phone numbers
+    for site in hpo_sites:
+        match_values = _compare_phone_numbers(
+            project,
+            validation_dataset,
+            ehr_dataset,
+            site,
+            consts.OBS_PII_PHONE,
+            consts.PHONE_NUMBER_FIELD
+        )
+        # write phone number matches for hpo to table
+        writers.append_to_result_table(
+            site,
+            match_values,
+            project,
+            validation_dataset,
+            consts.PHONE_NUMBER_FIELD
+        )
+
+    # validate genders
+    for site in hpo_sites:
+        match_values = _compare_genders(
+            project,
+            validation_dataset,
+            ehr_dataset,
+            site,
+            consts.OBS_PII_SEX
+        )
+        # write birthday match for hpo to table
+        writers.append_to_result_table(
+            site,
+            match_values,
+            project,
+            validation_dataset,
+            consts.SEX_FIELD
+        )
+
+    # validate birth dates
+    for site in hpo_sites:
+        match_values = _compare_birth_dates(
+            project,
+            validation_dataset,
+            ehr_dataset,
+            site,
+            consts.OBS_PII_BIRTH_DATETIME
+        )
+        # write birthday match for hpo to table
+        writers.append_to_result_table(
+            site,
+            match_values,
+            project,
+            validation_dataset,
+            consts.BIRTH_DATE_FIELD
+        )
+
+    # generate single clean record for each participant at each site
+    for site in hpo_sites:
+        writers.merge_fields_into_single_record(project, validation_dataset, site)
+        writers.remove_sparse_records(project, validation_dataset, site)
+        writers.change_nulls_to_missing_value(project, validation_dataset, site)
 
-    # TODO:  generate aggregate site report
+    # generate hpo site reports
+    for site in hpo_sites:
+        bucket = gcs_utils.get_hpo_bucket(site)
+        filename = os.path.join(
+            consts.REPORT_DIRECTORY.format(date=date_string),
+            consts.REPORT_TITLE
+        )
+        writers.create_site_validation_report(
+            project, validation_dataset, [site], bucket, filename
+        )
+
+    # generate aggregate site report
+    bucket = gcs_utils.get_drc_bucket()
+    filename = os.path.join(validation_dataset, consts.REPORT_TITLE)
+    writers.create_site_validation_report(
+        project, validation_dataset, hpo_sites, bucket, filename
+    )
 
     return results
 
 
 if __name__ == '__main__':
-    DATASET = 'combined20190211'
-    match_participants(DATASET)
+    RDR_DATASET = 'lrwb_combined20190415'   # the combined dataset
```

Comment:
these hard coded data set names should not be uploaded to the repository.

Reply:
I have removed the test variable names.